internal/parser/scanner/states.go

```diff
@@ -357,230 +570,526 @@ var keywordsWithW map[string]token.Type = map[string]token.Type{
 // }
 
 func scanAKeyword(s *scanner) token.Token {
-	input := string(s.input[s.start:s.seekNext(s.start)])
-	if _, ok := keywordsWithA[input]; ok {
-		s.acceptString(input)
-		return createToken(s.line, s.col, s.start, s.pos, keywordsWithA[input], input, s)
+	nextRune := s.seekNext(s.start)
```

Comment:
what does `seekNext` do?

Reply:
Returns the position of the end of this keyword.

Reply:
Then it should be named accordingly